tests/unit/test_spatial_methods/test_finite_volume/test_finite_volume.py

```diff
@@ -736,6 +754,36 @@ def test_definite_integral(self):
             integral_eqn_disc.evaluate(None, one_over_y), 2 * np.pi, decimal=3
         )
 
+        # cylindrical coordinates
+        mesh = get_cylindrical_mesh_for_testing(rcellpts=200)
+        spatial_methods = {"current collector": pybamm.FiniteVolume()}
+        disc = pybamm.Discretisation(mesh, spatial_methods)
+
+        var = pybamm.Variable("var", domain=["current collector"])
+        r = pybamm.SpatialVariable("r", ["current collector"])
+        integral_eqn = pybamm.Integral(var, r)
+        disc.set_variable_slices([var])
+        integral_eqn_disc = disc.process_symbol(integral_eqn)
+
+        pts = mesh["current collector"].nodes
+        r0 = mesh["current collector"].edges[0]
+        constant_y = np.ones_like(pts[:, np.newaxis])
+        np.testing.assert_array_almost_equal(
+            integral_eqn_disc.evaluate(None, constant_y), np.pi * (1 - r0 ** 2)
+        )
+        linear_y = pts
+        np.testing.assert_array_almost_equal(
+            integral_eqn_disc.evaluate(None, linear_y),
+            2 * np.pi / 3 * (1 - r0 ** 3),
+            decimal=4,
+        )
+        one_over_y = 1 / pts
+        np.testing.assert_array_almost_equal(
+            integral_eqn_disc.evaluate(None, one_over_y),
+            2 * np.pi * (1 - r0),
+            decimal=3,
+        )
+
```

Comment:
we really need to clean up this test file 😅

Reply:
Haha tell me about it. My favourite bit was the test that said it was testing div(grad(r^2))=6 but it was actually testing div(grad(6))=0...

There's no rush to merge this, so I'll wait, fix conflicts with #1810 and also have a go at tidying up these tests a bit.